File: backend/semgrep_rules/tests/python/kisa_sw_31_unsalted_hash.py
import hashlib
def unsafe(password):
    # ruleid: kisa.sw31.python.password-hash-without-salt
    return hashlib.sha256(password.encode()).hexdigest()
def safe(password, salt):
    # ok: kisa.sw31.python.password-hash-without-salt
    return hashlib.pbkdf2_hmac("sha256", password.encode(), salt, 200000)

File: backend/semgrep_rules/tests/javascript/kisa_sw_31_unsalted_hash.js
const crypto = require("crypto");
function unsafe(password) {
  // ruleid: kisa.sw31.javascript.password-hash-without-salt
  return crypto.createHash("sha256").update(password).digest("hex");
}
function safe(password, salt) {
  // ok: kisa.sw31.javascript.password-hash-without-salt
  return crypto.scryptSync(password, salt, 64);
}
